src/PrestaShopBundle/DependencyInjection/Compiler/LoadModulesAsBundlesPass.php

```diff
@@ -0,0 +1,174 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\Config\Resource\FileExistenceResource;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Config\FileLocator;
+
+/**
+ * Load services stored in modules.
+ */
+class LoadModulesAsBundlesPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $this->registerServicesFromModules($container);
+        $this->registerTemplatesFromModules($container);
+    }
+
+    /**
+     * Load all services registered in every module.
+     *
+     * @todo: only register services for installed modules.
+     *
+     * @param ContainerBuilder $container
+     * @return void
+     */
+    private function registerServicesFromModules(ContainerBuilder $container)
+    {
+        $modulesPaths = $this->getModulesPaths();
+
+        foreach($modulesPaths as $modulePath) {
+            if (file_exists($modulePath.'/config/services.yml')) {
+                $loader = new YamlFileLoader($container, new FileLocator($modulePath.'/config/'));
+                $loader->load('services.yml');
+            }
+        }
+    }
+
+    /**
+     * Allow the PrestaShop modules to override Twig templates
+     *
+     * @param ContainerBuilder $container
+     * @return void
+     */
+    private function registerTemplatesFromModules(ContainerBuilder $container)
+    {
+        $twigFilesystemLoaderDefinition = $container->getDefinition('twig.loader.filesystem');
+
+        $bundleHierarchy = $this->getBundleHierarchy($container);
+
+        foreach ($bundleHierarchy as $name => $bundle) {
+            $namespace = $this->normalizeBundleName($name);
+
+            foreach($this->getModulesPaths() as $modulePath) {
+                if (is_dir($dir = $modulePath.'/views/'.$namespace)) {
+                    $twigFilesystemLoaderDefinition->addMethodCall('addPath', array($dir, $namespace));
+                    /* Template from modules MUST be the first called if any */
+                    $twigFilesystemLoaderDefinition->setMethodCalls(array_reverse($twigFilesystemLoaderDefinition->getMethodCalls()));
```

Comment:
This will alternatively flip the order of templates if overridden more than once. Why not `unshift` instead?